api/common/v1alpha1/header.go

```diff
@@ -6,3 +7,31 @@ package v1alpha1
 type HeaderName string
 
 type HeaderValue string
+
+type HeaderMatchType string
+
+// HeaderMatchType constants.
+const (
+	HeaderMatchExact             HeaderMatchType = "Exact"
+	HeaderMatchPresent           HeaderMatchType = "Present"
+	HeaderMatchRegularExpression HeaderMatchType = "RegularExpression"
+	HeaderMatchAbsent            HeaderMatchType = "Absent"
+	HeaderMatchPrefix            HeaderMatchType = "Prefix"
+)
+
+// HeaderMatch describes how to select a HTTP route by matching HTTP request
+// headers.
+type HeaderMatch struct {
+	// Type specifies how to match against the value of the header.
+	// +optional
+	// +kubebuilder:default=Exact
+	// +kubebuilder:validation:Enum=Exact;Present;RegularExpression;Absent;Prefix
+	Type *HeaderMatchType `json:"type,omitempty"`
+
+	// Name is the name of the HTTP Header to be matched. Name MUST be lower case
+	// as they will be handled with case insensitivity (See https://tools.ietf.org/html/rfc7230#section-3.2).
+	Name HeaderName `json:"name"`
+
+	// Value is the value of HTTP Header to be matched.
+	Value HeaderValue `json:"value"`
```

Comment:
It seems like `Value` is optional, do we need `omitempty`?

Reply:
done